File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        ans = []
        def postorder(root,ans):
            if not root:
                return None
            postorder(root.left,ans)
            postorder(root.right,ans)
            ans.append(root.val)
        postorder(root,ans)
        return ans

File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void t(TreeNode* node, vector<int> &ans){
        if(node == nullptr){
            return;
        }
        //go left
        t(node->left,ans);
        //go right
        t(node->right,ans);
        //print data
        ans.push_back(node->val);
    }
    vector<int> postorderTraversal(TreeNode* root) {
        vector<int> ans;
        t(root,ans);
        return ans;
    }
};